### ISSUE TITLE
For UPI, show links to compatible wallets

### ISSUE THREAD

**Issue by remjx** [remjx/coindrop#272]
Per [@aahnik's request](https://github.com/remjx/coindrop/issues/268#issuecomment-778566740), "the coindrop site should not display a qr code. Instead it should give the option to open the link in any wallet app in their phone like Phone Pay, Google Pay, Paytm, etc."

**Comment by aahnik** [comment]
yup, thats for mobile. that will be great. I have seen that interface in UPI Gateway. https://upigateway.com/trial/page.

that is a paid platform, and generates a dynamic qr code, with the value specified. 

but in coindrop, when people will click and open in their wallet app, the person should have choice to type any value.

I did little research but have no idea how to get such links.

**Comment by aahnik** [comment]
Hi, @remjx , I think this feature request could be generalised for other payment methods. 

Like, when someone chooses the paypal option, from mobile, it should be able to open paypal app in phone. Similarly, for bitcoin it could pop up the list of compatible wallets installed in phone, to open with.

An extra question. Addresses are of two type. Like paypal has an web address like `paypal.me/user` or email of user, while btc has a btc address. The type of these two is different. Could there be option of choosing whether the address is an url or email or a crypto receiving address, or a upi address ?

for urls, there could be option of opening that link by clicking. but for btc or upi, a qr code to scan.

**Comment by remjx** [comment]
Currently, the "address" field is [handled](https://github.com/remjx/coindrop/blob/ada6adbb1456066ff9b76bde9044668eea30b8a7/components/PublicPiggybankPage/PaymentMethodButtonModal.tsx#L65) one of two ways: 

1. [Address is a valid url](https://github.com/remjx/coindrop/blob/ada6adbb1456066ff9b76bde9044668eea30b8a7/components/PublicPiggybankPage/PaymentMethodButtonModal.tsx#L22). In this case, the button says "Open" and will open in a new page. The destination webpage can handle opening the installed app. For example when I open https://paypal.me/mark in my mobile Chrome browser, it automatically opens the PayPal app (I'm on Android 10). Make sure you add the `https://` for the link to work. I'm not sure why I added that requirement, I will see if I can remove it so you can just enter `paypal.me/mark`.
2. Address is not a valid url. In this case, the button says "Copy" and will copy address to clipboard.

**Comment by aahnik** [comment]
thanks a lot for telling this feature. I had not used https://

**Comment by aahnik** [comment]
i did a little research and found out that UPI links (that open in upi compatible wallets in mobile, or display a QR when opened from desktop ) can be generated only for a fixed amount. 

There is no way to make a link, that will allow custom value to be choosen.
